test/src/test/java/jenkins/util/SystemPropertiesTest.java

```diff
@@ -61,10 +63,32 @@ public void shouldInitializeFromSystemProperty() throws Exception {
     }
     
     @Test
+    public void shouldInitializeFromWebAppProperty() throws Exception {
+        assertThat("Property is undefined before test",
+                SystemProperties.getString("foo.bar"), equalTo(null));
+        setWebAppInitParameter("foo.bar", "myVal");
+        assertThat("Web App property should assign the value",
+                SystemProperties.getString("foo.bar"), equalTo("myVal"));
+    }
+
+    @Test
     public void shouldUseSystemPropertyAsAHighPriority() throws Exception {
-        ServletContext c = j.jenkins.servletContext;
+        setWebAppInitParameter("install-wizard-path", "myVal1");
         System.setProperty("install-wizard-path", "myVal2");
         assertThat("System property should take system property with a high priority", 
                 SystemProperties.getString("install-wizard-path"), equalTo("myVal2"));
     }
+
+    /**
+     * Hack to set a web app initial parameter afterwards. Only works with Jetty.
+     * @param property property to set
+     * @param value value of the property
+     */
+    protected void setWebAppInitParameter(String property, String value) {
+        if (!(j.jenkins.servletContext instanceof ContextHandler.Context)) {
+            throw new IllegalArgumentException("This only works with Jetty");
```

Comment:
this is a test... better to use `Assume.assumeThat(j.jenkins.servletContext, Matchers.isInstance(ContextHandler.Context.class));` as that way if you run the tests on another servlet container they will be skipped rather than fail